org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeBuilder.java

```diff
@@ -1,31 +1,29 @@
 package org.code.javabuilder;
 
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.PrintStream;
+import java.io.*;
 import java.net.MalformedURLException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Comparator;
+import java.util.List;
 
 /** The orchestrator for code compilation and execution. */
 public class CodeBuilder implements AutoCloseable {
   private final OutputAdapter outputAdapter;
   private final InputAdapter inputAdapter;
-  private final ProjectFileManager fileManager;
   private final File tempFolder;
   private final PrintStream sysout;
   private final InputStream sysin;
+  private final UserProjectFiles userProjectFiles;
 
   public CodeBuilder(
-      InputAdapter inputAdapter, OutputAdapter outputAdapter, ProjectFileManager fileManager)
+      InputAdapter inputAdapter, OutputAdapter outputAdapter, UserProjectFiles userProjectFiles)
```

Comment:
This is a nice improvement!  Having CodeBuilder take a "data" object makes it more decoupled and easier to test and reuse.